Disable H8 extended battery info (_BIX) for all ThinkPad boards.

CONFIG_H8_HAS_BAT_INFO_EXTENDED enables the ACPI _BIX() method for
extended battery information, which on 26.09 exposes energy_* sysfs
files instead of charge_*. On some boards (e.g. X230) energy_now
reports 0, breaking battery display in Heads system info.

Disabling reverts to the charge_* interface used on 25.09/master.

Fixes: https://github.com/linuxboot/heads/issues/2222
diff --git a/src/mainboard/lenovo/haswell/Kconfig b/src/mainboard/lenovo/haswell/Kconfig
--- a/src/mainboard/lenovo/haswell/Kconfig
+++ b/src/mainboard/lenovo/haswell/Kconfig
@@ -7,7 +7,6 @@ config BOARD_LENOVO_HASWELL_COMMON
 	select EC_LENOVO_H8
 	select EC_LENOVO_PMH7
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_LEDLOGO
 	select H8_HAS_FNLOCK_LED
 	select H8_HAS_PRIMARY_FN_KEYS
diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig b/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
--- a/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
@@ -9,7 +9,6 @@ config BOARD_LENOVO_SKLKBL_THINKPAD_COMMON
 	select EC_LENOVO_MEC1653
 	select EC_LENOVO_PMH7
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_LEDLOGO
 	select H8_HAS_FNLOCK_LED
 	select H8_HAS_PRIMARY_FN_KEYS
diff --git a/src/mainboard/lenovo/t400/Kconfig b/src/mainboard/lenovo/t400/Kconfig
--- a/src/mainboard/lenovo/t400/Kconfig
+++ b/src/mainboard/lenovo/t400/Kconfig
@@ -14,7 +14,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_H8
 	select H8_HAS_2ND_THERMAL_ZONE
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION if !BOARD_LENOVO_R500
 	select BOARD_ROMSIZE_KB_8192 if !BOARD_LENOVO_R500
 	select BOARD_ROMSIZE_KB_4096 if BOARD_LENOVO_R500
diff --git a/src/mainboard/lenovo/t410/Kconfig b/src/mainboard/lenovo/t410/Kconfig
--- a/src/mainboard/lenovo/t410/Kconfig
+++ b/src/mainboard/lenovo/t410/Kconfig
@@ -25,7 +25,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select MAINBOARD_USES_IFD_GBE_REGION
 	select H8_HAS_2ND_THERMAL_ZONE
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select MAINBOARD_HAS_LIBGFXINIT
 	select DRIVERS_RICOH_RCE822
 	select DRIVER_LENOVO_SERIALS
diff --git a/src/mainboard/lenovo/t420/Kconfig b/src/mainboard/lenovo/t420/Kconfig
--- a/src/mainboard/lenovo/t420/Kconfig
+++ b/src/mainboard/lenovo/t420/Kconfig
@@ -12,7 +12,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_PMH7
 	select EC_LENOVO_H8
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION
 	select NO_UART_ON_SUPERIO
 	select BOARD_ROMSIZE_KB_8192
diff --git a/src/mainboard/lenovo/t420s/Kconfig b/src/mainboard/lenovo/t420s/Kconfig
--- a/src/mainboard/lenovo/t420s/Kconfig
+++ b/src/mainboard/lenovo/t420s/Kconfig
@@ -13,7 +13,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_PMH7
 	select GFX_GMA_PANEL_1_ON_LVDS
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION
 	select HAVE_ACPI_RESUME
 	select HAVE_ACPI_TABLES
diff --git a/src/mainboard/lenovo/t430s/Kconfig b/src/mainboard/lenovo/t430s/Kconfig
--- a/src/mainboard/lenovo/t430s/Kconfig
+++ b/src/mainboard/lenovo/t430s/Kconfig
@@ -12,7 +12,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_PMH7
 	select EC_LENOVO_H8
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION if BOARD_LENOVO_T430S
 	select H8_HAS_PRIMARY_FN_KEYS if BOARD_LENOVO_T431S
 	select NO_UART_ON_SUPERIO
diff --git a/src/mainboard/lenovo/t520/Kconfig b/src/mainboard/lenovo/t520/Kconfig
--- a/src/mainboard/lenovo/t520/Kconfig
+++ b/src/mainboard/lenovo/t520/Kconfig
@@ -10,7 +10,6 @@ config BOARD_LENOVO_BASEBOARD_T520
 	select EC_LENOVO_PMH7
 	select EC_LENOVO_H8
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION
 	select H8_HAS_WWAN_GPIO_DETECTION
 	select NO_UART_ON_SUPERIO
diff --git a/src/mainboard/lenovo/t530/Kconfig b/src/mainboard/lenovo/t530/Kconfig
--- a/src/mainboard/lenovo/t530/Kconfig
+++ b/src/mainboard/lenovo/t530/Kconfig
@@ -11,7 +11,6 @@ config BOARD_LENOVO_BASEBOARD_T530
 	select EC_LENOVO_PMH7
 	select GFX_GMA_PANEL_1_ON_LVDS
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION
 	select H8_HAS_WWAN_GPIO_DETECTION
 	select HAVE_ACPI_RESUME
diff --git a/src/mainboard/lenovo/x131e/Kconfig b/src/mainboard/lenovo/x131e/Kconfig
--- a/src/mainboard/lenovo/x131e/Kconfig
+++ b/src/mainboard/lenovo/x131e/Kconfig
@@ -25,7 +25,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select SOUTHBRIDGE_INTEL_C216
 	select SYSTEM_TYPE_LAPTOP
 	select USE_NATIVE_RAMINIT
-	select H8_HAS_BAT_INFO_EXTENDED
 
 config VBOOT
 	select GBB_FLAG_DISABLE_EC_SOFTWARE_SYNC
diff --git a/src/mainboard/lenovo/x1_carbon_gen1/Kconfig b/src/mainboard/lenovo/x1_carbon_gen1/Kconfig
--- a/src/mainboard/lenovo/x1_carbon_gen1/Kconfig
+++ b/src/mainboard/lenovo/x1_carbon_gen1/Kconfig
@@ -13,7 +13,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_PMH7
 	select GFX_GMA_PANEL_1_ON_LVDS
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION
 	select H8_HAS_WWAN_GPIO_DETECTION
 	select HAVE_ACPI_RESUME
diff --git a/src/mainboard/lenovo/x200/Kconfig b/src/mainboard/lenovo/x200/Kconfig
--- a/src/mainboard/lenovo/x200/Kconfig
+++ b/src/mainboard/lenovo/x200/Kconfig
@@ -13,7 +13,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_H8
 	select H8_HAS_2ND_THERMAL_ZONE if BOARD_LENOVO_X200
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION if BOARD_LENOVO_X200
 	select NO_UART_ON_SUPERIO
 	select BOARD_ROMSIZE_KB_8192
diff --git a/src/mainboard/lenovo/x201/Kconfig b/src/mainboard/lenovo/x201/Kconfig
--- a/src/mainboard/lenovo/x201/Kconfig
+++ b/src/mainboard/lenovo/x201/Kconfig
@@ -28,7 +28,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select MAINBOARD_USES_IFD_GBE_REGION
 	select H8_HAS_2ND_THERMAL_ZONE
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BDC_GPIO_DETECTION
 
 config VBOOT
diff --git a/src/mainboard/lenovo/x220/Kconfig b/src/mainboard/lenovo/x220/Kconfig
--- a/src/mainboard/lenovo/x220/Kconfig
+++ b/src/mainboard/lenovo/x220/Kconfig
@@ -13,7 +13,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_PMH7
 	select GFX_GMA_PANEL_1_ON_LVDS
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_WWAN_GPIO_DETECTION
 	select HAVE_ACPI_RESUME
 	select HAVE_ACPI_TABLES
diff --git a/src/mainboard/lenovo/x230/Kconfig b/src/mainboard/lenovo/x230/Kconfig
--- a/src/mainboard/lenovo/x230/Kconfig
+++ b/src/mainboard/lenovo/x230/Kconfig
@@ -12,7 +12,6 @@ config BOARD_SPECIFIC_OPTIONS
 	select EC_LENOVO_PMH7
 	select EC_LENOVO_H8
 	select H8_HAS_BAT_THRESHOLDS_IMPL
-	select H8_HAS_BAT_INFO_EXTENDED
 	select H8_HAS_BAT_CHARGE_BEHAVIOUR if BOARD_LENOVO_X230
 	select H8_HAS_PRIMARY_FN_KEYS if BOARD_LENOVO_X230S
 	select H8_HAS_WWAN_GPIO_DETECTION if !BOARD_LENOVO_X230S
